### DISCUSSION TITLE
生产环境与开发环境的配置

### DISCUSSION THREAD

**Discussion by panchanggang** [NervJS/taro#13176] [Q & A] [answered]
### 相关平台
微信小程序

**小程序基础库: 2.25.0**
**使用框架: React**

### 复现步骤
运行yarn dev:weapp之后，刚开始是cross-env NODE_ENV=development npm run build:weapp -- --watch； 但是之后又出现cross-env NODE_ENV=production taro build --type weapp "--watch"；

### 期望结果
运行yarn dev:weapp应该是将环境设置为开发环境，运行yarn build:weapp将环境设置为生产环境。

### 实际结果
运行yarn dev:weapp，第一行是将环境设置为开发环境，之后又出现了将环境设置为生产环境

### 环境信息
```
� Taro v3.4.11


  Taro CLI 3.4.11 environment info:
    System:
      OS: Windows 10
    Binaries:
      Node: 16.15.1 - E:\nodejs\node.EXE
      Yarn: 1.22.19 - E:\nodejs\node_global\yarn.CMD
      npm: 8.11.0 - E:\nodejs\npm.CMD
```

**Comment by panchanggang** [comment]
![image](https://user-images.githubusercontent.com/99158748/179334937-2b0bc794-6eab-4cbb-8a94-551e631044b3.png)
这个是截图

**Comment by TheKonka** [comment]
你自己package.json 里面写了， dev:weapp 里面执行了  build:weapp

**Comment by panchanggang** [comment]
那个应该怎么改呢？   这个之前是默认的，我没改过

**Comment by wangjunjia** [comment]
你这个 weapp 命令和其他命令不一样，参考着其他命令改。 @panchanggang

**Comment by panchanggang** [comment]
@wangjunjia   你好，这个是我自己设置的，为了区分开发环境和生产环境，但是我这样写出错了，不管运行哪个指令，他都是生产环境

**Comment by luchanan** [comment, marked as answer]
覆盖配置了吧，dev应用了build命令，另外起一行不行？

```
"scripts": {
    "build:h5": "cross-env DETAIL_ENV=dev taro build --type h5",
    "build:test:h5": "cross-env DETAIL_ENV=test taro build --type h5",
    "build:prod:h5": "cross-env DETAIL_ENV=prod taro build --type h5",
    "dev:h5": "npm run build:h5 -- --watch",
  },
```
